Addressing some comments received during 6MAN@IETF121

furry13 opened the pull request:
- Clarify the text about sending unsolicited RAs upon network config changes
-  Add more loss values to the table (0.7, 0.8, 0.9, 0.95)
- For the "valid lifetime < 2hrs" update:
  -- change the section to the "old text/new text format, so it's easier to read
  -- modify what text to remove (we need to delete more text from 4862)
  -- add some clarification about the threat model (as the original text contained it), as per Suresh's and Tim's comments.
  -- clarify the rationale to explain why the prefix needs to be invalidated.

furry13 committed:
Clarify what triggers out-of-schedule unsolicited RAs

Closes #5

furry13 committed:
Update draft-ietf-6man-slaac-renum-08.xml

Closes #6

We need to remove more text from Section 5.5.3 than just e), as all text after e) is actually talking about 2hrs lifetime rule, hence needs to be removed too.
So it makes sense to replace one justification with another, as Suresh suggested.

furry13 committed:
Add more loss values to the table

Closes #7

fgont reviewed (changes requested):
I'd probably tweak to "the information to be contained in" ("or to be sent in"..)

fgont commented on draft-ietf-6man-slaac-renum-08.xml:
```diff
@@ -360,18 +414,20 @@ This is to align the Router Lifetime with the recommendations in <xref target="R
-
-
+<t>
+While allowing updates to the valid lifetime in RAs could enable an attacker to invalidate addresses by setting the valid lifetime to zero, this doesn't significantly worsen the security situation. An attacker capable of sending rogue RAs already has the power to disrupt connectivity by manipulating other parameters, such as gateway or DNS information.  Therefore, accepting a zero lifetime doesn' ...
```
I'd probably:
* "doesn't" -> does not

* "accepting a zero lifetime doesn't make the system more vulnerable than it already is in the presence of such an attacker."  -> "accepting a zero lifetiem is one of the many vectors available to perform DoS attacks to local nodes"  and probably add an informative reference to rfc3756

furry13 replied:
> "doesn't" -> does not

Fixed.

>"accepting a zero lifetime doesn't make the system more vulnerable than it already is in the presence of such an attacker." ->  -> "accepting a zero lifetiem is one of the many vectors available to perform DoS attacks to local nodes" and probably add an informative reference to rfc3756

How about:

"accepting a zero lifetime doesn't make the system more vulnerable than it already is, as invalidating the prefix is just one of the many vectors available to perform DoS attacks to onlink nodes, see RFC3756"?

fgont replied:
I'm fine with the change. Two notes:

1) What I was meaning was to avoid "..more vulnerable" such that people don't argue "well, this is one more vector... even if at the end of the day having an extra vector doesn't make a practical difference, since there are so many!"

2) I'd put "..to onlink nodes (see [RFC3756])" within parenthesis..but you probably meant that, anyway. :-)

furry13 replied:
> 2. I'd put "..to onlink nodes (see [RFC3756])" within parenthesis..but you probably meant that, anyway. :-)

Yep, fixed, thanks!

fgont commented on draft-ietf-6man-slaac-renum-08.xml:
```diff
@@ -360,18 +414,20 @@ This is to align the Router Lifetime with the recommendations in <xref target="R
+</t>
+<t>
+In scenarios where RA-based attacks are of concern, proper mitigations such as RA-Guard <xref target="RFC6105"/> <xref target="RFC7113"/> or SEND <xref target="RFC3971"/> should be implemented. 
+</t>
```
I'd probaby s/proper//

-- the thing here is that SEND is probably not much deplyable. And, if you care about DoS, RA-Guard will hopefully mitigate the forged RA packets. But then an attacker can still forge e.g. NAs ane.g. map the router's link-local IPv6 address to a non-ecisting link-layer address, for DoS purposes.  (stricly speaking RA-Guard will mitigate "RA-based attacks", as noted, though, but not ND attacks, which may also employ other ND messages...)

furry13 replied:
> I'd probaby s/proper//

Done.

As this paragraph is about a specific RA-based attack, I think it's OK not to focus on ND-based.
We may mention p2p isolation as another approach, but I think it is not exactly in scope for this paragraph...

fgont replied:
Focusing on RA-based attacks is fair enough, as you note.  (the threat model for ND is that you trust onlink nodes and ND packets, anyway).

fgont commented on draft-ietf-6man-slaac-renum-08.xml:
```diff
@@ -360,18 +414,20 @@ This is to align the Router Lifetime with the recommendations in <xref target="R
 <list style="symbols">
 <t>
-This change allows hosts to react to the signal provided by a router that has positive knowledge that a prefix has become invalid.
+This change allows hosts to react to the signal provided by a router that has positive knowledge that a prefix is not assigned to the given link anymore. In particular it would allow the host to invalidate addresses from that prefix, and, if the prefix is reassigned to another link, allows the host to communicate to devices on that link.
```
FWIW, just to note that I like this one! -- the text is better than the original ;=)

furry13 replied:
;) Glad to hear ;)

fgont reviewed (changes requested):
Other than the small twaeks, looks great!

fgont reviewed (changes requested):
I'd probably rephrase as:
"
When an interface is initialized or the information contained in RAs changes (e.g. an interface is reconfigured)..
"

-- i.e., this seems like two cases: interface initialization and config changes (only the later possibly resulting from reconfiguraiton)

fgont commented on draft-ietf-6man-slaac-renum-08.xml:
FWIW, I'm fine with the OLD/TEXT/NEW-TEXT. -- FWIW, my take was that essentially the rest of the text (items 1.3 plus footnote discussing the attacks) was part of the footnote (as per their indentation). -- but probably spelling out OLD/NEW clarifies things... so the change improves on that.  (@JanZorz ? @r-pats ? )

furry13 replied:
The reason I prefer 'OLD/NEW TEXT' is that I, when reading the doc and encounter smth like "replace the last paragraph of Section X of RFC12345 with the following text" need to open another browser window with the RFC in question and start comparing the text. Having everything in one page makes the text more readable, IMHO.

>my take was that essentially the rest of the text (items 1.3 plus footnote discussing the attacks) was part of the footnote (as per their indentation)

I had to print the doc as pdf to see what you mean (indentation is clearly not my thing, that' why I can't write Python ;)) - there are multiple paragraphs there, so it's easy to be confused, IMHO - so for readability I think we shall make it explicit what we delete.

fgont replied:
This is also fair enough -- my comment was just a clarification of what the existing text "meant" -- but any changings that help clarity and make things easier for the reader are always a good thing. (thanks for catching this!)

furry13 commented:
> I'd probably tweak to "the information to be contained in" ("or to be sent in"..)

OK, I've replaced:
"When the information contained in RAs changes" with "When the information to be contained in RAs changes"

furry13 committed:
Addressing Fernando's comments

furry13 commented:
> I'd probably rephrase as: " When an interface is initialized or the information contained in RAs changes (e.g. an interface is reconfigured).. "
> 
> -- i.e., this seems like two cases: interface initialization and config changes (only the later possibly resulting from reconfiguraiton)

Yes, I was thinking about that, but the whole section is about sending RAs *upon changes*, not when the interface is initialized (the latter is already covered, and we are not changing that).
Am I missing smth?

fgont commented:
@furry13 : I took a look at the text again (orig draft, and rfc4862), and it seems there are a few issues:

The title of *our* section is "Interface Initialization". However, the text in our section only references Section 6.2.4 of RFC4862 (that has a title of "Sending Unsolicited Router Advertisements", and this section of RFC4862 talks a bit both about RAs that are sent upon interface initialization, and RAs that are sent when information changes -- but only has normative language about the later

It would seem to me that we should do the following:

1. Changing the title of our section to "Sending Unsolicited Router Advertisements" (but see below before applying this change :-) )

2. Our text should probably address two things (but it currently does just one): recommending to send unsolicited RAs upon information changes, *and* sending RAs upon interface initialization (which it currently does not). I skimmed throughRFC4862, but failed to see a "MUST" (or SHOULD/MAY) that requires unsolicited RAs to be sent upon interface initialization or when the interface becomes and advertising interface.  If there's lready such a MUST, then our text should only focus on information changes. OTOH, if there's currently no "MUST" about this in RFC4862 (i.e., there's not RFC2119 requirement about this, or it's only a MAY/SHOULD), then our doc should probably have text on the topic.

(from what i've seen in RFC4862, the only *implicit* requirement to send unsolicited RAs upon interface initialization is the maximum allowed time between unsolicited RAs -- but that's not enough -- a pathological implementation might wait for such time before sending RAs, rather than sending unsolicited RAs right after the interface has been initialized (modulo the RA sending rules, of course))

Thoughts?

furry13 committed:
Add missing parenthesis

furry13 commented:
> @furry13 : I took a look at the text again (orig draft, and rfc4862), and it seems there are a few issues:
> 
> The title of _our_ section is "Interface Initialization". However, the text in our section only references Section 6.2.4 of RFC4862 (that has a title of "Sending Unsolicited Router Advertisements", and this section of RFC4862 talks a bit both about RAs that are sent upon interface initialization, and RAs that are sent when information changes -- but only has normative language about the later
>
> (from what i've seen in RFC4862, the only _implicit_ requirement to send unsolicited RAs upon interface initialization is the maximum allowed time between unsolicited RAs -- but that's not enough -- a pathological implementation might wait for such time before sending RAs, rather than sending unsolicited RAs right after the interface has been initialized (modulo the RA sending rules, of course))

> It would seem to me that we should do the following:
> 
> 1. Changing the title of our section to "Sending Unsolicited Router Advertisements" (but see below before applying this change :-) )
> 2. Our text should probably address two things (but it currently does just one): recommending to send unsolicited RAs upon information changes, _and_ sending RAs upon interface initialization (which it currently does not). I skimmed throughRFC4862, but failed to see a "MUST" (or SHOULD/MAY) that requires unsolicited RAs to be sent upon interface initialization or when the interface becomes and advertising interface.  If there's lready such a MUST, then our text should only focus on information changes. OTOH, if there's currently no "MUST" about this in RFC4862 (i.e., there's not RFC2119 requirement about this, or it's only a MAY/SHOULD), then our doc should probably have text on the topic.

I've re-read 4861, and I believe there is implict requirements:
- Section 6.2.2 "Becoming an Advertising Interface" says that an advertizing interface is an interface which has AdvSendAdvertisements = true (hence the router sends periodic RAs from that interface). 
- so the question is "when the first RA is sent, when an interface becomes an advertizing interface"?
- the answer is in the paragraph just before the one we are changing:
"For the first few advertisements (up to
   MAX_INITIAL_RTR_ADVERTISEMENTS) sent from an interface when it
   becomes an advertising interface, if the randomly chosen interval is
   greater than MAX_INITIAL_RTR_ADVERT_INTERVAL, the timer SHOULD be set
   to MAX_INITIAL_RTR_ADVERT_INTERVAL instead.  Using a smaller interval
   for the initial advertisements increases the likelihood of a router
   being discovered quickly when it first becomes available, in the
   presence of possible packet loss."

So it clearly says that upon becoming an advertizing interface, the router SHOULD send the first MAX_INITIAL_RTR_ADVERTISEMENTS (3) RAs with MAX_INITIAL_RTR_ADVERT_INTERVAL (16 secs) or shorter intervals between them.

> 
> Thoughts?



> @furry13 : I took a look at the text again (orig draft, and rfc4862), and it seems there are a few issues:
> 
> The title of _our_ section is "Interface Initialization". However, the text in our section only references Section 6.2.4 of RFC4862 (that has a title of "Sending Unsolicited Router Advertisements", and this section of RFC4862 talks a bit both about RAs that are sent upon interface initialization, and RAs that are sent when information changes -- but only has normative language about the later
> 
> It would seem to me that we should do the following:
> 
> 1. Changing the title of our section to "Sending Unsolicited Router Advertisements" (but see below before applying this change :-) )
> 2. Our text should probably address two things (but it currently does just one): recommending to send unsolicited RAs upon information changes, _and_ sending RAs upon interface initialization (which it currently does not). I skimmed throughRFC4862, but failed to see a "MUST" (or SHOULD/MAY) that requires unsolicited RAs to be sent upon interface initialization or when the interface becomes and advertising interface.  If there's lready such a MUST, then our text should only focus on information changes. OTOH, if there's currently no "MUST" about this in RFC4862 (i.e., there's not RFC2119 requirement about this, or it's only a MAY/SHOULD), then our doc should probably have text on the topic.
> 
> (from what i've seen in RFC4862, the only _implicit_ requirement to send unsolicited RAs upon interface initialization

Yes, it's implicit, but I think it's rather clear and can't be misread - advertiing interface is one which the router sends RAs from. How often - also specified.

> is the maximum allowed time between unsolicited RAs

No, the timer is much shorter.
The paragraph above the one we are changing is saying, that the first 3 RAs SHOULD be sent with no more than MAX_INITIAL_RTR_ADVERT_INTERVAL (16 seconds) between them. So the max waiting period is 16 seconds.

> -- but that's not enough -- a pathological implementation might wait for such time before sending RAs, rather than sending unsolicited RAs right after the interface has been initialized (modulo the RA sending rules, of course))
> Thoughts?

I have preference to focus on advertising *changes*.

Sending RAs immediately upon interface initialization does not help with renumerbing (as the change here is from "not being a router" to "being a router", so the host has no stale info about that interface yet.

What we *might* want to do is to discuss if we want to reduce MAX_INITIAL_RTR_ADVERT_INTERVAL. 
BTW I did suggest sending RAs *immediately* upon the config change (when I presented https://www.ietf.org/archive/id/draft-link-6man-truce-00.html - which was basically just one change we are dsicussing now). People were against "immidiate" in case multiple interfaces are coming up at the same time (e.g. a power outage).
Also, "immidiate" is not a great thing if the link flaps. 
Currently the changes will be propagated in 16 seconds. Do we want to change it to..5?

furry13 committed:
Renaming 'Interface Initialization'

..and moving it up in the doc

furry13 commented:
So, following our discussion, I've renamed 'Interface initialization' to '[Propagating Interface Configuration Changes' and moved the section up, so it's not right after 'Propagating stale info' section - looks more logical to me.

how does it look? Shall we merge?

fgont commented:
HI, Jen,

Will do a final review today.

Thnanx!



On 2/2/25 21:37, Jen wrote:
> So, following our discussion, I've renamed 'Interface initialization' to 
> '[Propagating Interface Configuration Changes' and moved the section up, 
> so it's not right after 'Propagating stale info' section - looks more 
> logical to me.
> 
> how does it look? Shall we merge?
> 
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/furry13/slaac-renum-revive/pull/9#issuecomment-2629658460>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAOBON6LFSCIU2K23SU6NLL2N226PAVCNFSM6AAAAABVRY2PJCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMMRZGY2TQNBWGA>.
> You are receiving this because you were assigned.Message ID: 
> ***@***.***>
> 

-- 
Fernando Gont
SI6 Networks
e-mail: ***@***.***
PGP Fingerprint: F242 FF0E A804 AF81 EB10 2F07 7CA1 321D 663B B494

r-pats commented:
LGTM

fgont committed:
Expand the text that needs to be replaced

Clarify that information may also change as a result of it being derived from information learned via an upstream router.
i.e., spell out out that the flash-renum scenario is one of the possible causes for the information change, other than
"system management".
